GitUI/LeftPanel/RepoObjectsTree.cs

```diff
@@ -16,7 +16,7 @@
 
 namespace GitUI.LeftPanel
 {
-    public sealed partial class RepoObjectsTree : GitModuleControl
+    public sealed partial class RepoObjectsTree : GitModuleControl, IScriptHostControl
```

Comment:
As the RevisionGrid is always injected into the ROT, the ROT itself does not need to *be* an / implement IScriptHostControl. Rather turn `_scriptHostControl` into a property.

Reply:
It needs to implement `IScriptHostControl` because it passes itself as a `Control & IScriptHostControl`. C# doesn't have intersection types, so this is the next best option. If the parameter was just `IScriptHostControl`, then `_scriptHostControl` could be passed directly.

Or do you mean it should, instead, implement an interface that _provides_ an `IScriptHostControl` instead of being it? It's a possibility, but all it does is add one extra layer of indirection, which then needs to be applied to `RevisionGridControl` as well.

Reply:
There is no need to "pass itself" - at least not in `RepoObjectsTree.ContextActions.cs`, where `_scriptHostControl` could be passed directly.

Reply:
`_scriptHostControl` can't be passed directly, because `_scriptHostControl` is `IScriptHostControl`, and the parameter is `THostControl where THostControl : Control, IScriptHostControl`. In other words, passing `_scriptHostControl` would cause a compile error. Casting it to `THostControl` is impossible, because intersection types cannot be defined (and there's a hella long discussion about it). And the parameter can't be changed to not be `THostControl`, because then `hostControl.FindForm()` would be impossible. It's all connected.